Rustdoc does not obey --error-format for doctests error messages

I tried this code:

```rust
/// ```rust
/// foo();
/// ```
fn foo() {
    println!("Hello, world!");
}
```

I expected to see this happen: Running `cargo test --doc --message-format=short` uses short messages for all errors

Instead, this happened: The build failure for the doc test emits full error messages:

```console
> cargo test --doc --message-format=short
   Compiling foo v0.1.0 (/tmp/tmp.RZdiSTk0tC/foo)
src/lib.rs:4:4: warning: function is never used: `foo`
warning: 1 warning emitted
    Finished test [unoptimized + debuginfo] target(s) in 0.11s
   Doc-tests foo

running 1 test
test src/lib.rs - foo (line 1) ... FAILED

failures:

---- src/lib.rs - foo (line 1) stdout ----
error[E0423]: expected function, found crate `foo`
 --> src/lib.rs:2:1
  |
4 | foo();
  | ^^^ not a function

error: aborting due to previous error

For more information about this error, try `rustc --explain E0423`.
Couldn't compile the test.

failures:
    src/lib.rs - foo (line 1)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.04s

error: test failed, to rerun pass '--doc'
```

### Meta

`rustc --version --verbose`:
```
rustc 1.50.0-nightly (0f6f2d681 2020-12-06)
binary: rustc
commit-hash: 0f6f2d681b39c5f95459cd09cb936b6ceb27cd82
commit-date: 2020-12-06
host: x86_64-unknown-linux-gnu
release: 1.50.0-nightly
```

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"poliorcetics"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/src/librustdoc/doctest.rs
+++ b/src/librustdoc/doctest.rs
@@ -296,7 +296,12 @@ fn run_test(
         }
     });
     if let ErrorOutputType::HumanReadable(kind) = options.error_format {
-        let (_, color_config) = kind.unzip();
+        let (short, color_config) = kind.unzip();
+
+        if short {
+            compiler.arg("--error-format").arg("short");
+        }
+
         match color_config {
             ColorConfig::Never => {
                 compiler.arg("--color").arg("never");

--- a/src/test/rustdoc-ui/issue-81662-shortness.rs
+++ b/src/test/rustdoc-ui/issue-81662-shortness.rs
@@ -1,0 +1,12 @@
+// compile-flags:--test --error-format=short
+// normalize-stdout-test: "src/test/rustdoc-ui" -> "$$DIR"
+// normalize-stdout-test "finished in \d+\.\d+s" -> "finished in $$TIME"
+// failure-status: 101
+
+/// ```rust
+/// foo();
+/// ```
+//~^^ ERROR cannot find function `foo` in this scope
+fn foo() {
+    println!("Hello, world!");
+}

--- a/src/test/rustdoc-ui/issue-81662-shortness.stdout
+++ b/src/test/rustdoc-ui/issue-81662-shortness.stdout
@@ -1,0 +1,16 @@
+
+running 1 test
+test $DIR/issue-81662-shortness.rs - foo (line 6) ... FAILED
+
+failures:
+
+---- $DIR/issue-81662-shortness.rs - foo (line 6) stdout ----
+$DIR/issue-81662-shortness.rs:7:1: error[E0425]: cannot find function `foo` in this scope
+error: aborting due to previous error
+Couldn't compile the test.
+
+failures:
+    $DIR/issue-81662-shortness.rs - foo (line 6)
+
+test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in $TIME
+

